LibraryApp/src/components/books-list.js

```diff
@@ -0,0 +1,240 @@
+import React from 'react';
+import {
+  StyleSheet,
+  View,
+  Text,
+  TextInput,
+  SafeAreaView,
+  ActivityIndicator,
+  Pressable,
+} from 'react-native';
+import {connect} from 'react-redux';
+import {setActiveTab, setSelectedBook} from 'store/actions/library';
+import {Colors, Global, Spacing, Typography} from 'stylesheets';
+import {libraryForm} from 'constants/app-defaults';
+import {FilterListBySearch} from 'helpers/library';
+import List from 'components/shared/list';
+import {Icons} from 'constants/icons';
+
+const loaderTimeout = 2000;
+const {search} = Icons;
+
+const {
+  flex1,
+  rowFlex,
+  borderRadius20,
+  horizontalCenter,
+  verticalCenter,
+} = Global;
+const {comicFont, gochiFont, iconFont, fs18, fs20, fs25, bold} = Typography;
+const {p10, py20, px10, px15, px25, p15, mtAuto, mb15} = Spacing;
+const {blue, grey, darkGrey, purple, lightGreen, maroon, white} = Colors;
+
+class BooksList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isDataFetching: false,
+      isRefreshing: false,
+      searchText: '',
+    };
+    this.onRefresh = this.onRefresh.bind(this);
+    this.getApiData = this.getApiData.bind(this);
+    this.selectBook = this.selectBook.bind(this);
+    this.search = this.search.bind(this);
+    this.addNewBook = this.addNewBook.bind(this);
+  }
+
+  componentDidMount() {
+    const {books} = this.props;
+    if (books.length == 0) {
+      this.setState({
+        isDataFetching: true,
+      });
+    }
+    setTimeout(() => {
+      this.setState({isDataFetching: false});
+    }, loaderTimeout);
+  }
+
+  onRefresh() {
+    this.setState({isRefreshing: true}, () => {
+      this.getApiData();
+    });
+  }
+
+  getApiData() {
+    setTimeout(() => {
+      this.setState({isRefreshing: false});
+    }, loaderTimeout);
+  }
+
+  selectBook(book) {
+    const {setSelectedBook} = this.props;
+    setSelectedBook(book);
+    this.navigateToForm();
+  }
+
+  renderItem = ({item}) => {
+    const {name, author, publisher, price} = item;
+    const {listItemContainer, listItemName, bookPrice} = styles;
+    return (
+      <Pressable
+        style={listItemContainer}
+        onPress={() => this.selectBook(item)}>
+        <View style={[flex1, px15]}>
+          <Text style={listItemName}>{name}</Text>
+          <Text style={[gochiFont, fs20]}>{author}</Text>
+          <Text style={[gochiFont, fs18]}>{publisher}</Text>
+        </View>
+        <View>
+          <Text style={bookPrice}>{price}</Text>
+        </View>
+      </Pressable>
+    );
+  };
+
+  getItemKey = (item) => `book-${item.id}`;
+
+  search(value) {
+    this.setState({
```

Comment:
this is inhibiting user behaviour.Search should be trimmed in serach logic not here

Reply:
Done